build.rs

```diff
@@ -1,11 +1,57 @@
 // Copyright The ocicrypt Authors.
 // SPDX-License-Identifier: Apache-2.0
 
+use std::fs::File;
+use std::io::{Read, Write};
+use ttrpc_codegen::Customize;
+
 fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let protos = vec!["src/utils/proto/keyprovider.proto"];
     #[cfg(feature = "keywrap-keyprovider")]
     tonic_build::configure()
         .build_server(true)
-        .out_dir("src/utils/")
-        .compile(&["src/utils/proto/keyprovider.proto"], &["src/utils"])?;
+        .out_dir("src/utils/grpc")
+        .compile(&protos, &["src/utils"])?;
+
+    ttrpc_codegen::Codegen::new()
```

Comment:
Can we add a new feature for ttrpc support? As a keyprovider client, we may need support only one type with build config.